Card/src/main/java/com/braintreepayments/api/AuthenticationInsight.java

```diff
@@ -13,33 +16,30 @@
     private static final String GRAPHQL_REGULATION_ENVIRONMENT_KEY = "customerAuthenticationRegulationEnvironment";
     private static final String REST_REGULATION_ENVIRONMENT_KEY = "regulationEnvironment";
 
-    private String regulationEnvironment;
+    private final String regulationEnvironment;
 
     static AuthenticationInsight fromJson(JSONObject json) {
         if (json == null) {
             return null;
         }
 
-        AuthenticationInsight authInsight = new AuthenticationInsight();
         String regulationEnv;
-
         if (json.has(GRAPHQL_REGULATION_ENVIRONMENT_KEY)) {
-            regulationEnv = Json.optString(json, GRAPHQL_REGULATION_ENVIRONMENT_KEY, null);
+            regulationEnv = Json.optString(json, GRAPHQL_REGULATION_ENVIRONMENT_KEY, "");
         } else {
-            regulationEnv = Json.optString(json, REST_REGULATION_ENVIRONMENT_KEY, null);
+            regulationEnv = Json.optString(json, REST_REGULATION_ENVIRONMENT_KEY, "");
         }
 
         if ("psdtwo".equalsIgnoreCase(regulationEnv)) {
             regulationEnv = "psd2";
         }
+        regulationEnv = regulationEnv.toLowerCase();
 
-        if (regulationEnv != null ) {
-            regulationEnv = regulationEnv.toLowerCase();
-        }
-
-        authInsight.regulationEnvironment = regulationEnv;
+        return new AuthenticationInsight(regulationEnv);
+    }
 
-        return authInsight;
+    AuthenticationInsight(String regulationEnvironment) {
+        this.regulationEnvironment = regulationEnvironment;
```

Comment:
Moving `regulationEnvironment` property to the constructor allows us to make the property `@NonNull` on AuthenticationInsight.